hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban/AzkabanHelper.groovy

```diff
@@ -412,19 +412,36 @@ class AzkabanHelper {
   /**
    * Reads the Azkaban session ID from the session.properties file in the ~/.azkaban directory.
    *
+   * @param azkabanUrl Azkaban URL
+   *
    * @return The Azkaban session ID
    */
-  static String readSession() {
-    File file = new File(System.getProperty("user.home") + "/.azkaban/session.properties");
-    String sessionId = null;
+  static String readSession(String azkabanUrl) {
+    String path = System.getProperty("user.home") + "/.azkaban/" + filesystemFriendly(azkabanUrl) + ".properties"
+    File file = new File(path);
+    logger.lifecycle("Searching for properties on path " + path + "\n");
     if (file.exists()) {
+      String sessionId = null;
+      long sessionTimestamp = 0
```

Comment:
The assumption of Azkaban's session TTL is unnecessary as the validation is being done by individual instances calling the readSession() method.  If the session's expired the user is asked to enter the password which is already taken care of.

Can you please make changes to remove the use of timestamp and validation for it in here?

Overall, this PR compliments #110 for users using multiple azkaban instances concurrently. I'm looking forward for your changes :)

Reply:
Hi @pranayhasan! The sole purpose of the `sessionTimestamp` is to put upper limit on the age of the `sessionId`. If a session is retrieved by the plugin on say Friday and then the plugin is used again on Monday, the `sessionId` is already considered expired and not used anymore.
Use case: this comes very handy on slow networks during the execution of the `azkabanUpload` task: consider that the Azkaban zip is high tens of megabytes in size and we want to upload it after more then a day of no communication with Azkaban. If we try to use sessionId indiscriminately, then we could try to upload a huge file, tens of minutes later this upload would predictably fail (since the sessionId from long ago is not valid anymore), then we would retrieve a fresh sessionId, and then the upload would finally succeed after considerable amount of time. This is not practical.

Reply:
Your mentioned use-case is already addressed in #90
Although your mentioned way invalidates an unnecessary call in few cases, I think it's better to keep it simple :)
Let me know if you're not convinced as we can take @nntnag17 's opinion.

Reply:
Hi @pranayhasan ! I am not strongly opposed to removing it, it just makes more sense to me that a sessionId is accompanied by a timestamp. Session to me means a token+lifespan. From the practical impact: it makes difference on a slow line, and the overhead of checking the validity of the token is low.
Sure, let's hear @nntnag17 's or anybody else's opinion to get extra insight. :)

Reply:
Hi! I think the benefit that we get from adding a timestamp is not huge since #90 already makes a lightweight request to check if sessionId is valid. Also, as a client, we should not make assumptions about the session lifespan which is governed by the server. IMHO, it's better not to add the timestamp to the session id and let #90 handle session validation.

Reply:
Thanks @nntnag17